drivers/sensor/nxp/p3t1755/p3t1755.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_p3t1755
+
+#include "p3t1755.h"
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+#include <stdlib.h>
+
+LOG_MODULE_REGISTER(P3T1755, CONFIG_SENSOR_LOG_LEVEL);
+
+static int p3t1755_read_reg(const struct device *dev, uint8_t reg, uint8_t *value, uint8_t len)
+{
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	struct p3t1755_data *data = dev->data;
+
+	return i3c_burst_read(data->i3c_dev, reg, value, len);
+#else
```

Comment:
Will this work if there are two instances of the sensor in the system, one on an I3C bus and one on an I2C bus? The ifdef checks if *any* instance of the sensor is on an I3C bus, not if *this* instance of the sensor is on an I3C bus.

Reply:
Thanks Maureen, I have made a change to use function pointers and assigning it on a per instance basis.